### ISSUE TITLE
tool manifest should contain source

### ISSUE THREAD

**Issue by orcun** [dotnet/cli#12289]
## Steps to reproduce
Install a tool locally using a local source

## Expected  behavior
dotnet-tools.json to include source information, beside version, commands, etc.

## Actual behavior
dotnet-tools.json does not record the source of the package

This would help to update local tools with different sources.

## Environment data
Version: 3.0.0-preview8-28405-07

**Comment by wli3** [comment]
@orcun thank you for the issue. What is the scenario you want to achieve? nuget.config (to specify source) can be store with your source code as well. For example https://github.com/NuGet/NuGet.Client/blob/dev/NuGet.Config

**Comment by orcun** [comment]
Perhaps generating nuget.config or an entry if it exits would probably work. I am not sure if that also supports file paths but assuming that it does, how can naming conflicts be resolved? I would also prefer to read the source of the package right next to it and not search for it.

**Comment by orcun** [comment]
You are absolutely right. I didn't even check if it were also the case for package references before posting this issue. That is something I was expecting by default because of experience with other package managers in many different languages and I assumed this was probably due to new local tool manifest support.
